docs/389ds/FAQ/bdb-deprecation.md

```diff
@@ -0,0 +1,277 @@
+
+There are several ways:
+
+1. Using dscreate --interactive:
+   Select the proper implementation when the following question is asked:
+
+   ```
+   Choose whether mdb or bdb is used.
+   ```
+
+2. Using dscreate --from-file
+   Set the following stanza in the template file:
+
+   ```
+   [slapd]
+   db_lib = mdb (or bdb)
+   ```
+
+3. Using dscreate with an environment variable.
+   Setting NSSLAPD_DB_LIB environment variable to 'bdb' or 'mdb' change the backend implementation chosen by default
+
+Set  db_lib = mdb (or bdb) in [slapd] section of dscreate template file
+
+## Can I mix different implementations ?
+
+Mixing implementaions limitations are:
+
+- The db_lib parameter is stored in the nsslapd-backend-implement attribute of cn=config entry, so it is global to the instance.
+- backup / restore formats are tied to the db implementation so you cannot use a backup to restore an instance with a different implementation. (ldif export/import and initializing through the replication have not this limitation)
+
+| Question                                                                    | Answer |
+| --------------------------------------------------------------------------- | ------ |
+| Can I mix backends with different implementations in an instance ?          | No     |
+| Can I mix instances with different backends implementations in an host      | Yes    |
+| Can I mix replicas with different implementation in a replicated topology ? | Yes    |
+
+## How to migrate between database implementation ?
+
+There are several ways to do that.
+
+### Automatic method
+
+- dsctl *instanceName* dblib bdb2mdb
+
+- dsctl *instanceName* dblib mdb2bdb
+
+Note: these tools does not cleanup the old data (so can revert by simply changing nsslapd-backend-implement back).
+To remove the old database, after verifying that everything is working as expected, the following command should be run:
+
+- dsctl *instanceName* dblib cleanup
+
+#### Manual Method - export to ldif
+
+[A] Export all the backends to ldif
+
+[B] Determine the lmdb database maximum size. A way to do that could be
+to compute the existing databases size and add a security margin:
+
+- Add the results of
+
+  ```
+    du -hs  /var/lib/dirsrv/slapd-instanceName/db/*/.
+  ```
+
+  - Add a 20% Margin
+
+[C] Change nsslapd-backend-implement:
+
+```
+dsconf instanceName backend config set --db-lib mdb
+dsctl instanceName restart
+```
+
+[D] Set mdb-max-size to the value computed in step [B]
+
+```
+dsconf instanceName backend config set --mdb-max-size 2G
+dsctl instanceName restart
+```
+
+[E] Import the ldif on all backends
+
+### Manual method - initialize through replication
+
+Same than the export method with the following changes:
+
+skip step [A]
+
+replace step [E] by: initialize all backends through replication from another supplier
+
+## What is the plan for OS that does not anymore provide libdb ?
+
+We plan to provide a tool that will still be able to read the bdb databases to
```

Comment:
Should we open the upstream ticket for that tool and add it to the document ?

Reply:
My feeling is that we should let the FAQ evolve when the tools is ready